interp/src/primitives/stateful.rs

```diff
@@ -234,6 +192,32 @@ impl StdDivPipe {
 }
 
 impl Primitive for StdDivPipe {
+    fn do_tick(&mut self) -> Vec<(ir::Id, OutputValue)> {
+        if let Some(Some((q, r))) = self.queue.pop_back() {
+            self.queue.push_front(self.update.take());
+            assert_eq!(
+                self.queue.len(),
+                3,
+                "std_div_pipe's internal queue has length {} != 3",
+                self.queue.len()
+            );
+            vec![
+                (ir::Id::from("out_quotient"), q.into()),
+                (ir::Id::from("out_remainder"), r.into()),
+                (ir::Id::from("done"), Value::bit_high().into()),
+            ]
+        } else {
+            self.queue.push_front(self.update.take());
+            assert_eq!(
+                self.queue.len(),
+                3,
+                "std_div_pipe's internal queue has length {} != 3",
+                self.queue.len()
+            );
+            vec![]
+        }
+    }
+
```

Comment:
Same comment about this seeming to take 4 cycles (unless I've misunderstood the inner workings of the vec-deqeue which is definitely possible